### DISCUSSION TITLE
Questions about accuracy and speed

### DISCUSSION THREAD

**Discussion by joiemoie** [sign-language-translator/sign-language-translator#34] [General]
Hi! I stumbled on this code looking for an app I can build for deaf people. I was wondering if you could help with some questions I have about this code:

1) Which sign languages does this support?
2) Does this run on GPU?
3) What is the speed and runtime? Could this be run in realtime?
4) Do you have any data on the accuracy of this?

Thanks!

**Comment by mdsrqbl** [comment]
Hi. This project is still under development but I hope the following helps you out:

1. In this Iteration, the package has the implementation of only one sign language class which is built around the data obtained from various institutes in Pakistan about their "standardized" sign language. The package can also serve as a framework to implement custom sign language classes which can easily integrate into its rule-based translator. Please let me know If the documentation is not sufficient. Deep learning models can be multilingual and can handle out-of-vocabulary words but they are not available yet ([roadmap](https://github.com/sign-language-translator/sign-language-translator/tree/landmarks_wrapper?tab=readme-ov-file#upcomingroadmap)).
2. The PyTorch models are very easy to shift to any device of your choice and defaults to CUDA if it is available. The non-deep learning based models don't support batching and will be slower. All implementation is purely python based so we do not get the C/C++ speedup most libraries offer. The video file wrapper uses OpenCV and using a GPU based OpenCV compilation is an enhancement somewhere down the line.
3. Although this package only has a rule-based translation model at the moment (you can test its speed on [HuggingFace Spaces](https://huggingface.co/spaces/sltAI/ConcatenativeSynthesis)), I don't think you will be able to run sign to text translation in real time in today's common hardware. The devices most people own are not powerful enough to run large models and even if the models are deployed on server machines, the users may not be privileged with an internet bandwidth fast enough to stream videos to the models.
4. I will work on evaluating and improving the translation quality once the first iteration is complete and the deep learning pipeline has shipped some working translation models.

Hope this helps.
